src/stream/src/common/table/state_table.rs

```diff
@@ -1293,6 +1293,22 @@ where
     }
 }
 
+impl<
+        S,
+        SD,
+        const IS_REPLICATED: bool,
+        W: WatermarkBufferStrategy,
+        const USE_WATERMARK_CACHE: bool,
+    > StateTableInner<S, SD, IS_REPLICATED, W, USE_WATERMARK_CACHE>
+where
+    S: StateStore,
+    SD: ValueRowSerde,
+{
+    pub fn get_schema_len(&self) -> usize {
+        self.pk_indices.len() + self.value_indices.as_ref().map_or(0, |v| v.len())
+    }
+}
+
```

Comment:
Modified state table to add a `get_schema_len` function, which obtains the schema length with value + pk indices.
Is there some better way to get schema length?

I'm using it for backwards compatibility.

This is done by comparing expected row length (pk_indices + 3), with the actual schema length, which could be `pk_indices + 2` OR `pk_indices + 3 (after this PR)`.

Reply:
For more context, the schema was previously:

`| vnode | pk_indices ... | backfill_is_finished |`

Now it's:

`| vnode | pk_indices ... | backfill_is_finished | row_count |`

Reply:
So by comparing it, we know whether we need to handle row count.

Reply:
Do we need to handle backward compatibility in this case? I mean for the original backfill states, they must be finished already so we don't need to access the `row_count` field.

Reply:
> Do we need to handle backward compatibility in this case? I mean for the original backfill states, they must be finished already so we don't need to access the `row_count` field.

Yes, because we need to query the backfill state and access the `backfill_is_finished` column, to know if backfill is finished.
The column's position has changed because we added a new `row_count` field.

Reply:
Oh I guess for checking the row_count, we can just see if that column is present in the state. If it isn't we just treat it as finished already.

Reply:
> hey must be finished already so we don't need to access the row_count field.
Makes sense.

Reply:
Resolved https://github.com/risingwavelabs/risingwave/pull/12415/commits/d50acbacb282bf1ec5782e2c425143883d78b9bf